File: GeneratedEntityReaderClasses/Classes/ValueRangeInt.cs
namespace GeneratedNoitaClasses;
public class ValueRangeInt :  iNoitaType<ValueRangeInt>{
	public NoitaInt min { get; set; } = new();
	public NoitaInt max { get; set; } = new();

    public void Read(NoitaStream s){
		min.Read(s);
		max.Read(s);

    }
    public void Write(NoitaStream s){
		min.Write(s);
		max.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/ValueRangeInt.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class ValueRangeInt:
    min:NoitaInt = field(default_factory=NoitaInt)
    max:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.min.Read(s)
        self.max.Read(s)
    
    def Write(self, s:NoitaStream):
        self.min.Write(s)
        self.max.Write(s)

